--- a/mysql-test/r/information_schema.result	2009-05-07 19:31:26.000000000 +0000
+++ b/mysql-test/r/information_schema.result	2009-05-07 19:32:59.000000000 +0000
@@ -60,6 +60,7 @@
 USER_STATISTICS
 VIEWS
 INNODB_IO_PATTERN
+INNODB_RSEG
 columns_priv
 db
 func
@@ -743,7 +744,7 @@
 CREATE VIEW a1 (t_CRASHME) AS SELECT f1 FROM t_crashme GROUP BY f1;
 CREATE VIEW a2 AS SELECT t_CRASHME FROM a1;
 count(*)
-109
+110
 drop view a2, a1;
 drop table t_crashme;
 select table_schema,table_name, column_name from
@@ -819,7 +820,7 @@
 flush privileges;
 SELECT table_schema, count(*) FROM information_schema.TABLES GROUP BY TABLE_SCHEMA;
 table_schema	count(*)
-information_schema	24
+information_schema	25
 mysql	17
 create table t1 (i int, j int);
 create trigger trg1 before insert on t1 for each row
@@ -1228,6 +1229,7 @@
 USER_STATISTICS	USER
 VIEWS	TABLE_SCHEMA
 INNODB_IO_PATTERN	SPACE
+INNODB_RSEG	RSEG_ID
 SELECT t.table_name, c1.column_name
 FROM information_schema.tables t
 INNER JOIN
@@ -1267,6 +1269,7 @@
 USER_STATISTICS	USER
 VIEWS	TABLE_SCHEMA
 INNODB_IO_PATTERN	SPACE
+INNODB_RSEG	RSEG_ID
 SELECT MAX(table_name) FROM information_schema.tables;
 MAX(table_name)
 VIEWS
@@ -1342,6 +1345,7 @@
 INDEX_STATISTICS	information_schema.INDEX_STATISTICS	1
 INNODB_BUFFER_POOL_CONTENT	information_schema.INNODB_BUFFER_POOL_CONTENT	1
 INNODB_IO_PATTERN	information_schema.INNODB_IO_PATTERN	1
+INNODB_RSEG	information_schema.INNODB_RSEG	1
 KEY_COLUMN_USAGE	information_schema.KEY_COLUMN_USAGE	1
 PROCESSLIST	information_schema.PROCESSLIST	1
 PROFILING	information_schema.PROFILING	1
--- a/mysql-test/r/information_schema_db.result	2009-05-07 19:31:27.000000000 +0000
+++ b/mysql-test/r/information_schema_db.result	2009-05-07 19:35:01.000000000 +0000
@@ -29,6 +29,7 @@
 USER_STATISTICS
 VIEWS
 INNODB_IO_PATTERN
+INNODB_RSEG
 show tables from INFORMATION_SCHEMA like 'T%';
 Tables_in_information_schema (T%)
 TABLES
--- a/mysql-test/r/mysqlshow.result	2009-05-07 19:31:26.000000000 +0000
+++ b/mysql-test/r/mysqlshow.result	2009-05-07 19:36:32.000000000 +0000
@@ -103,6 +103,7 @@
 | USER_STATISTICS                       |
 | VIEWS                                 |
 | INNODB_IO_PATTERN                     |
+| INNODB_RSEG                           |
 +---------------------------------------+
 Database: INFORMATION_SCHEMA
 +---------------------------------------+
@@ -132,6 +133,7 @@
 | USER_STATISTICS                       |
 | VIEWS                                 |
 | INNODB_IO_PATTERN                     |
+| INNODB_RSEG                           |
 +---------------------------------------+
 Wildcard: inf_rmation_schema
 +--------------------+
--- a/mysql-test/r/profiling.result	2009-05-28 19:39:42.000000000 +0000
+++ b/mysql-test/r/profiling.result	2009-05-28 19:40:14.000000000 +0000
@@ -6,6 +6,8 @@
 Variable_name	Value
 profiling	OFF
 profiling_history_size	15
+profiling_server	OFF
+profiling_use_getrusage	OFF
 select @@profiling;
 @@profiling
 0
@@ -16,12 +18,16 @@
 Variable_name	Value
 profiling	OFF
 profiling_history_size	100
+profiling_server	OFF
+profiling_use_getrusage	OFF
 set session profiling = ON;
 set session profiling_history_size=30;
 show session variables like 'profil%';
 Variable_name	Value
 profiling	ON
 profiling_history_size	30
+profiling_server	OFF
+profiling_use_getrusage	OFF
 select @@profiling;
 @@profiling
 1
--- a/mysql-test/r/mysql.result 2010-02-19 23:59:36.000000000 -0500
+++ b/mysql-test/r/mysql.result 2010-02-19 23:58:50.000000000 -0500
@@ -162,8 +162,8 @@
 ERROR 1049 (42000) at line 1: Unknown database 'invalid'
 Test connect with dbname + hostname
 Test connect with dbname + _invalid_ hostname
-ERROR 2005 (HY000) at line 1: Unknown MySQL server host 'invalid_hostname' (errno)
-ERROR 2005 (HY000) at line 1: Unknown MySQL server host 'invalid_hostname' (errno)
+ERROR 2003 (HY000) at line 1: Can't connect to MySQL server on 'invalid_hostname' (errno)
+ERROR 2003 (HY000) at line 1: Can't connect to MySQL server on 'invalid_hostname' (errno)
 The commands reported in the bug report
 ERROR 2005 (HY000) at line 1: Unknown MySQL server host 'cyril has found a bug :)XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' (errno)
 Too long dbname
@@ -198,6 +198,6 @@
 1
 COUNT (*)
 1
-ERROR 2005 (HY000) at line 1: Unknown MySQL server host 'invalid_hostname' (errno)
+ERROR 2003 (HY000) at line 1: Can't connect to MySQL server on 'invalid_hostname' (errno)
 <TABLE BORDER=1><TR><TH>&lt;</TH></TR><TR><TD>&lt; &amp; &gt;</TD></TR></TABLE>
 End of 5.0 tests
--- a/mysql-test/r/mysql_upgrade.result 2010-02-19 23:58:16.000000000 -0500
+++ b/mysql-test/r/mysql_upgrade.result 2010-02-20 00:01:34.000000000 -0500
@@ -58,7 +58,7 @@
 mysql.user                                         OK
 DROP USER mysqltest1@'%';
 Run mysql_upgrade with a non existing server socket
-mysqlcheck: Got error: 2005: Unknown MySQL server host 'not_existing_host' (errno) when trying to connect
+mysqlcheck: Got error: 2003: Can't connect to MySQL server on 'not_existing_host' (errno) when trying to connect
 FATAL ERROR: Upgrade failed
 set GLOBAL sql_mode='STRICT_ALL_TABLES,ANSI_QUOTES,NO_ZERO_DATE';
 mysql.columns_priv                                 OK
diff -Nur a/mysql-test/std_data/cacert.pem b/mysql-test/std_data/cacert.pem
--- a/mysql-test/std_data/cacert.pem	2010-02-20 00:28:37.000000000 -0500
+++ b/mysql-test/std_data/cacert.pem	2010-02-20 00:29:28.000000000 -0500
@@ -1,17 +1,22 @@
 -----BEGIN CERTIFICATE-----
-MIICrTCCAhagAwIBAgIJAJXpePU0UOTVMA0GCSqGSIb3DQEBBQUAMEQxCzAJBgNV
+MIIDsjCCApqgAwIBAgIJALpTYwUml0btMA0GCSqGSIb3DQEBBQUAMEQxCzAJBgNV
 BAYTAlNFMRAwDgYDVQQIEwdVcHBzYWxhMRAwDgYDVQQHEwdVcHBzYWxhMREwDwYD
-VQQKEwhNeVNRTCBBQjAeFw0wOTAxMjgxMDQ5NDZaFw0xNDAxMjcxMDQ5NDZaMEQx
+VQQKEwhNeVNRTCBBQjAeFw0xMDAyMjAwNTIyMjdaFw0xMjExMTYwNTIyMjdaMEQx
 CzAJBgNVBAYTAlNFMRAwDgYDVQQIEwdVcHBzYWxhMRAwDgYDVQQHEwdVcHBzYWxh
-MREwDwYDVQQKEwhNeVNRTCBBQjCBnzANBgkqhkiG9w0BAQEFAAOBjQAwgYkCgYEA
-4XQHAe5R1+TXC8noZtWf+d5E0v1C59FWpn9SWEUCBjE5UiIwuJvi4Y+7xWGOXLAI
-/JzJx5gNXLBiTsE/zh0uX9fKlajLhxB0GN+QU0ZlpQ1BeYipEcNXeI/7cT499f6v
-XWabnTflivdCgHSWUOQ20/Lzs6kP6/e6OoZd/DPSjPECAwEAAaOBpjCBozAdBgNV
-HQ4EFgQU8uLqVWWkmuKsnZf1RWz294wRrd8wdAYDVR0jBG0wa4AU8uLqVWWkmuKs
-nZf1RWz294wRrd+hSKRGMEQxCzAJBgNVBAYTAlNFMRAwDgYDVQQIEwdVcHBzYWxh
-MRAwDgYDVQQHEwdVcHBzYWxhMREwDwYDVQQKEwhNeVNRTCBBQoIJAJXpePU0UOTV
-MAwGA1UdEwQFMAMBAf8wDQYJKoZIhvcNAQEFBQADgYEAMMTE5sDN+Z0ZlV7KvH3g
-6+aKvql8dTpRT3hYukeQlWua0nq74WPGVw0c4e/M/vbiMwmJcCYpB9pd4+dHqzSw
-aPyoenjY6UF8n7B4quWy3SIUk2LSHeJLW+kzJn2afN9gvipFhdVh/uU2TIyLGOur
-Z/vmJX2W7hF1uqPnbfa8Lrw=
+MREwDwYDVQQKEwhNeVNRTCBBQjCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoC
+ggEBAL5nMws7aqf57iD/ZQUshASGnH/lpAYW5tVxMsmfiASTsDDbyO2bKxpGfFgl
+cRFTsUxsVcSnn78W7XY709HJhOoM3k1LkAg2gt2VdGoQxrVhYQRUjR67Tu+5ecR/
+768C+0/kUrhxZAtc+mmoWBwrUpGGyhTZfK9GiaI669p1oa1HeSBxjCc+jcxTQ3B9
+XK0RpsRyeYpE3tsfPowxZi6k9+Gim186ekaLBkwmKrwLZK6KTRIkpiZMrtJCMlSg
+bF/897xUJASDvswIp7PkIcZ/82Fk0MyUhqkwyifQjWdGY+B+1700Q74l5KzEMbHT
+Xl2RnIkP/1M8snFMUZYAAaTmAhcCAwEAAaOBpjCBozAdBgNVHQ4EFgQUT9RwITO/
+Ai1VWc7PbH/CuAxa7TAwdAYDVR0jBG0wa4AUT9RwITO/Ai1VWc7PbH/CuAxa7TCh
+SKRGMEQxCzAJBgNVBAYTAlNFMRAwDgYDVQQIEwdVcHBzYWxhMRAwDgYDVQQHEwdV
+cHBzYWxhMREwDwYDVQQKEwhNeVNRTCBBQoIJALpTYwUml0btMAwGA1UdEwQFMAMB
+Af8wDQYJKoZIhvcNAQEFBQADggEBAK4PVJXdTdVkmA9bTFnUELi1MqQ4wjgQesCP
+Jubh1I7fk1y1wJyQC76b/ARGTu+EA2SnHRbgWJuqrQZvn/6qRZbDfT/lnqnqHPn1
+7tHQOM9BdmMgFBn44QoUt/fLLdtOdHN/N8Y0i4yZGX5i5Xq1kXVh2ApHzvSEFnc3
+XQGHn+KAcIWMTCBtQCXpz1UBWXTiqxKTtP6wd0xYTb0c7eTL/PNMXQkopcU/EgYe
+m2P7q9AdEBha1vGRMoxxT5htra/a3dZ0P9p4y2WkZX910x+HuJpr8yqHSgvuO4o2
+Dx4nTo3SYuqn/M6ZPSpzVSiVwd+tP/WmW2gstMI4x89wevSDKoY=
 -----END CERTIFICATE-----
diff -Nur a/mysql-test/std_data/client-cert.pem b/mysql-test/std_data/client-cert.pem
--- a/mysql-test/std_data/client-cert.pem	2010-02-20 00:28:37.000000000 -0500
+++ b/mysql-test/std_data/client-cert.pem	2010-02-20 00:29:48.000000000 -0500
@@ -1,55 +1,19 @@
-Certificate:
-    Data:
-        Version: 3 (0x2)
-        Serial Number: 3 (0x3)
-        Signature Algorithm: sha1WithRSAEncryption
-        Issuer: C=SE, ST=Uppsala, L=Uppsala, O=MySQL AB
-        Validity
-            Not Before: Jan 28 11:04:39 2009 GMT
-            Not After : Jan 28 11:04:39 2010 GMT
-        Subject: C=SE, ST=Uppsala, O=MySQL AB/emailAddress=abstract.mysql.developer@mysql.com
-        Subject Public Key Info:
-            Public Key Algorithm: rsaEncryption
-            RSA Public Key: (512 bit)
-                Modulus (512 bit):
-                    00:e1:52:30:2c:d9:be:64:28:91:5d:7a:fd:d9:e9:
-                    14:35:7a:d2:94:4e:91:46:e0:db:9f:6b:79:f4:4c:
-                    ac:6e:07:61:34:86:74:62:a7:a8:44:af:fa:87:87:
-                    a8:7d:42:61:ff:ab:50:d4:7b:bf:75:fa:d5:d5:b3:
-                    74:fb:56:1e:37
-                Exponent: 65537 (0x10001)
-        X509v3 extensions:
-            X509v3 Basic Constraints: 
-                CA:FALSE
-            X509v3 Subject Key Identifier: 
-                58:30:B5:9B:2C:05:94:06:BA:3D:3C:F0:B2:CD:1D:67:65:E3:7F:85
-            X509v3 Authority Key Identifier: 
-                keyid:F2:E2:EA:55:65:A4:9A:E2:AC:9D:97:F5:45:6C:F6:F7:8C:11:AD:DF
-                DirName:/C=SE/ST=Uppsala/L=Uppsala/O=MySQL AB
-                serial:95:E9:78:F5:34:50:E4:D5
-
-    Signature Algorithm: sha1WithRSAEncryption
-        05:19:e3:13:14:fc:c5:28:bf:69:f8:00:b3:25:cb:bd:ca:9f:
-        2f:4c:b3:a8:04:11:f0:74:27:bd:82:2c:b4:49:9b:a7:59:f0:
-        f7:87:d1:e0:ba:99:a2:fe:4b:1d:10:6f:e4:a2:b3:cd:7f:8b:
-        68:31:46:ee:cd:9e:e2:47:e1:4c:fa:74:d1:e2:8b:cc:a0:4b:
-        a8:24:d1:a4:c3:6b:2a:c6:28:cd:41:e0:06:48:e6:cf:f2:3c:
-        ca:37:95:d7:29:64:6b:91:91:83:e7:ac:c8:0b:87:bc:da:a6:
-        aa:f1:44:43:c8:74:7b:15:26:91:2e:03:c4:71:50:6c:f8:68:
-        dc:8c
 -----BEGIN CERTIFICATE-----
-MIICfzCCAeigAwIBAgIBAzANBgkqhkiG9w0BAQUFADBEMQswCQYDVQQGEwJTRTEQ
-MA4GA1UECBMHVXBwc2FsYTEQMA4GA1UEBxMHVXBwc2FsYTERMA8GA1UEChMITXlT
-UUwgQUIwHhcNMDkwMTI4MTEwNDM5WhcNMTAwMTI4MTEwNDM5WjBlMQswCQYDVQQG
-EwJTRTEQMA4GA1UECBMHVXBwc2FsYTERMA8GA1UEChMITXlTUUwgQUIxMTAvBgkq
-hkiG9w0BCQEWImFic3RyYWN0Lm15c3FsLmRldmVsb3BlckBteXNxbC5jb20wXDAN
-BgkqhkiG9w0BAQEFAANLADBIAkEA4VIwLNm+ZCiRXXr92ekUNXrSlE6RRuDbn2t5
-9EysbgdhNIZ0YqeoRK/6h4eofUJh/6tQ1Hu/dfrV1bN0+1YeNwIDAQABo4GjMIGg
-MAkGA1UdEwQCMAAwHQYDVR0OBBYEFFgwtZssBZQGuj088LLNHWdl43+FMHQGA1Ud
-IwRtMGuAFPLi6lVlpJrirJ2X9UVs9veMEa3foUikRjBEMQswCQYDVQQGEwJTRTEQ
-MA4GA1UECBMHVXBwc2FsYTEQMA4GA1UEBxMHVXBwc2FsYTERMA8GA1UEChMITXlT
-UUwgQUKCCQCV6Xj1NFDk1TANBgkqhkiG9w0BAQUFAAOBgQAFGeMTFPzFKL9p+ACz
-Jcu9yp8vTLOoBBHwdCe9giy0SZunWfD3h9Hgupmi/ksdEG/korPNf4toMUbuzZ7i
-R+FM+nTR4ovMoEuoJNGkw2sqxijNQeAGSObP8jzKN5XXKWRrkZGD56zIC4e82qaq
-8URDyHR7FSaRLgPEcVBs+GjcjA==
+MIIDHTCCAgUCAQEwDQYJKoZIhvcNAQEFBQAwRDELMAkGA1UEBhMCU0UxEDAOBgNV
+BAgTB1VwcHNhbGExEDAOBgNVBAcTB1VwcHNhbGExETAPBgNVBAoTCE15U1FMIEFC
+MB4XDTEwMDIyMDA1MjYxMVoXDTEyMTExNjA1MjYxMVowZTELMAkGA1UEBhMCU0Ux
+EDAOBgNVBAgTB1VwcHNhbGExETAPBgNVBAoTCE15U1FMIEFCMTEwLwYJKoZIhvcN
+AQkBFiJhYnN0cmFjdC5teXNxbC5kZXZlbG9wZXJAbXlzcWwuY29tMIIBIjANBgkq
+hkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxWChFsS8pEj8CrU6aqiojdfeff1Q5S3I
+qc/B4opnmuGfPW4sszKgG2r6XJcc0YpFNapvXdUf1IWQ+Et9CcZVnPTcO8RDv3jj
+2I/cWWDkdOdRflxadvOEtSfLQorSj3s/sfHCuizeI3a/TFnvOEZD5Mv8efN1vDM9
+/dGjdY9Ck12aGeFwdALPU9J9ofdtsfy5fsu6KT9etVD2YIXa+Yeo7OiQIPtpmjUv
+EnNy7ZeCkNnG3Jd3MqxbtajfjsUPS+gwZ61R7hZnn/5SzQTkXIrpDlP1FOlC/41J
+dxmz9cskHhf+dqpOw0nrxZSBFL3hx/s8eHj27YU4WEwN02cQOEweMwIDAQABMA0G
+CSqGSIb3DQEBBQUAA4IBAQAnpYN2x0a9S7RxCfNx5H/IOiGwtdj3chIXNU2VJkoz
+JrqXSa9g0pUDbl4sBBkwEY5HkFx8Ed514Di1nq6qRKv/aaCJyHa2aQNDyCEccUOr
+YFh1OyHpO6kj5nei6syJXVvyRfJjequ+S/dmjHrH9YW1g5gT06CWeT1Z/tLOqi4f
+JsaUkCMA6+Oy6sVctWABjKSMIzQYV1fI8X+HQAncoaoFhABxS4pHWnsklGmiIOdK
+nab+6M3ITmHgW6rnub/CMymHcVFjoEh58A5mc3FAA1wzPwCkGimNHTE615+GngT0
+ccT5Ia79ieSWz65qkLzAkWOHhJK9IdzLg6aEBOY5u15S
 -----END CERTIFICATE-----
diff -Nur a/mysql-test/std_data/client-key.pem b/mysql-test/std_data/client-key.pem
--- a/mysql-test/std_data/client-key.pem	2010-02-20 00:28:37.000000000 -0500
+++ b/mysql-test/std_data/client-key.pem	2010-02-20 00:29:48.000000000 -0500
@@ -1,9 +1,27 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIIBOQIBAAJBAOFSMCzZvmQokV16/dnpFDV60pROkUbg259refRMrG4HYTSGdGKn
-qESv+oeHqH1CYf+rUNR7v3X61dWzdPtWHjcCAwEAAQJAXYooM8ZlcuEgj+VKU1ee
-qyEFIMqJJxqcMk+E/nWCM96WxCP3zHNSrqNfSpI3ld7QzMwhdRz+gFLxT2gGNpIw
-MQIhAPxzM/lDihe67X3ADYtDl9ZjA8Pm430x9sXlcxI17tCZAiEA5H1SyFl4mUee
-9VnfSC2XGW7lwz72ZygfVX+b7tLWF08CIEh40gzW5MfXM+KLxdea+fXjyursV5ZT
-R6KcMiKiNQLRAiAcmHqlzFzFgisotai2Fc6VRkXHG7gmzOSvBJt1VjmpDQIge6jf
-2N7whTdvC4ferB+zUlgWQdyvx1c3T4gnt6PYdaY=
+MIIEpAIBAAKCAQEAxWChFsS8pEj8CrU6aqiojdfeff1Q5S3Iqc/B4opnmuGfPW4s
+szKgG2r6XJcc0YpFNapvXdUf1IWQ+Et9CcZVnPTcO8RDv3jj2I/cWWDkdOdRflxa
+dvOEtSfLQorSj3s/sfHCuizeI3a/TFnvOEZD5Mv8efN1vDM9/dGjdY9Ck12aGeFw
+dALPU9J9ofdtsfy5fsu6KT9etVD2YIXa+Yeo7OiQIPtpmjUvEnNy7ZeCkNnG3Jd3
+MqxbtajfjsUPS+gwZ61R7hZnn/5SzQTkXIrpDlP1FOlC/41Jdxmz9cskHhf+dqpO
+w0nrxZSBFL3hx/s8eHj27YU4WEwN02cQOEweMwIDAQABAoIBACN2dBsCKB6gWmCW
+JTPKr9YkZf5fCCzVKAuSn3RTh7/lL9Y6uF9OBkRG2UrPf8cKP9sfjzjdtQVbJivm
+bEXYU8OKnxNB9SYu00kyt9V7OjDi6tp4P2zF3h8qUjPXe3N6InyXQnRu4U6f5EkU
+k42Z2lL+b5N9sbapwkabZvIVVlHhm6PEQTDOyvdr93zNmpWU3HvZ8pnt2/mKs7fa
+hrquQkcsx1X+BWxyiElPAjvX8BfCQGweH51yD38F6nzKH5RBszPjMkM5+pkKz+QG
+M365k/M9Vq4uPaSiZgZBy57cQ0CS/jin9CjS0YEF9CX7UE2x8QSaobZcf8WmOXff
+7g8+UXECgYEA5sWQ0G/zigpqfJ9g0N5bWYbhHkWg5emld15mOoAMUjmVsByy2ZNb
+ifQcTBuXqUDJkbL1xujCCnDEPYgIVxWVe3SJE+ntD/6A1RNWIy+u7LvbNcBhjwT1
+p2FSLDMFiIiW7I0v7zCmC/3FxRU3dqHXSxWho0oU7nMesgPoK1oTY+kCgYEA2vR7
+psETeLiichcF2k5ex+WDokMNSacFiqY9qQOlXWi1HIbfrNLID/UGsr+aUeURh6le
+pTA/CxZhU/uMjigp45fXsGLnRnoLDE5jkCES+cK8XRJJe/SBAhnInroOanPz2FPr
+BHCB3TfwBB1HRsxCBFUWMdplLRXP047DnmFKK7sCgYEAhcv0CzVtQRKK7NhG/LJ+
+yihOawu5nrNZcRz1LmPBQQcMB4CMHWO3RgOXSHUS+FmMxIG6x+l5Xy5ESNyO4Vat
+RVeqhjYxcP2A2okMfvgNdknwbXit14rrlnDxgJjQU3nF7z5olUhibT9kI8EKWCi7
+N1hCT/cKrblnCZUETeY6qckCgYEA066J3S3NJWlbo/GeNvCNsbSzDqzEq/BN6z3N
+XxmnRid08Z9zoCIoSUUatLWI/KP5zq5wUkPSr2Vzsz3/JXCvDto+LMlxzzEufTdM
+CovvTIh4k4KaiLxZ/S3wR4SZa1Gu7fYzwJcE9lqeMVwdDNVs/cNFMe1y46OyyVW8
+yeJFUwECgYBiXHQfivjBVKpUNaYBsg8VjpScwJkHOti/zfO3AGxxXRCGpq2ElQzl
+ISKgiVierHg8zUEzrgGKv+bzXNJrCfh3dDxwm9OUSWbImRSJiguxEQUeTcGA+Pm3
+I4ZI2dzIdh07RuB0vAIwqFaNbiZ6KEWjrE9/MxtpcQsfpAu0QFcQsw==
 -----END RSA PRIVATE KEY-----
diff -Nur a/mysql-test/std_data/server-cert.pem b/mysql-test/std_data/server-cert.pem
--- a/mysql-test/std_data/server-cert.pem	2010-02-20 00:28:37.000000000 -0500
+++ b/mysql-test/std_data/server-cert.pem	2010-02-20 00:30:02.000000000 -0500
@@ -1,55 +1,20 @@
-Certificate:
-    Data:
-        Version: 3 (0x2)
-        Serial Number: 1 (0x1)
-        Signature Algorithm: sha1WithRSAEncryption
-        Issuer: C=SE, ST=Uppsala, L=Uppsala, O=MySQL AB
-        Validity
-            Not Before: Jan 28 10:55:13 2009 GMT
-            Not After : Jan 28 10:55:13 2010 GMT
-        Subject: C=SE, ST=Uppsala, O=MySQL AB, CN=localhost/emailAddress=abstract.mysql.developer@mysql.com
-        Subject Public Key Info:
-            Public Key Algorithm: rsaEncryption
-            RSA Public Key: (512 bit)
-                Modulus (512 bit):
-                    00:b6:8f:e5:b7:b4:86:83:13:8a:f9:bf:63:cb:64:
-                    2d:b9:51:d1:de:ab:7b:45:1f:aa:b5:66:73:13:f9:
-                    a6:07:d5:ba:7c:fa:92:bd:37:e2:ad:87:db:3e:b6:
-                    6a:12:64:f8:ee:17:e3:15:06:2f:a8:82:68:bf:57:
-                    8d:c3:04:98:27
-                Exponent: 65537 (0x10001)
-        X509v3 extensions:
-            X509v3 Basic Constraints: 
-                CA:FALSE
-            X509v3 Subject Key Identifier: 
-                D9:9A:B8:5F:22:EA:04:10:C8:25:7D:82:57:E6:2E:FD:19:29:E7:DA
-            X509v3 Authority Key Identifier: 
-                keyid:F2:E2:EA:55:65:A4:9A:E2:AC:9D:97:F5:45:6C:F6:F7:8C:11:AD:DF
-                DirName:/C=SE/ST=Uppsala/L=Uppsala/O=MySQL AB
-                serial:95:E9:78:F5:34:50:E4:D5
-
-    Signature Algorithm: sha1WithRSAEncryption
-        54:07:2d:21:0b:a5:af:3b:58:23:32:5e:56:7f:ab:58:63:48:
-        91:aa:38:90:89:16:f9:cc:bf:a4:0e:78:2b:9f:c5:1b:58:a6:
-        e6:08:8f:2e:ae:97:03:21:9b:f1:cd:c0:26:8f:1d:d7:28:27:
-        a0:8e:81:09:1b:1c:0f:c9:a5:41:3a:2d:44:3f:9c:fa:87:ff:
-        c8:4c:2b:44:f7:1b:c1:3e:4f:01:7f:e9:26:cc:9f:1c:06:b5:
-        0b:27:d1:10:90:be:93:0c:9c:e7:b0:d1:ea:27:99:4e:06:14:
-        0c:7a:e9:c1:52:c5:33:68:bc:61:0d:db:81:3b:57:48:57:bf:
-        42:9a
 -----BEGIN CERTIFICATE-----
-MIICkzCCAfygAwIBAgIBATANBgkqhkiG9w0BAQUFADBEMQswCQYDVQQGEwJTRTEQ
-MA4GA1UECBMHVXBwc2FsYTEQMA4GA1UEBxMHVXBwc2FsYTERMA8GA1UEChMITXlT
-UUwgQUIwHhcNMDkwMTI4MTA1NTEzWhcNMTAwMTI4MTA1NTEzWjB5MQswCQYDVQQG
-EwJTRTEQMA4GA1UECBMHVXBwc2FsYTERMA8GA1UEChMITXlTUUwgQUIxEjAQBgNV
-BAMTCWxvY2FsaG9zdDExMC8GCSqGSIb3DQEJARYiYWJzdHJhY3QubXlzcWwuZGV2
-ZWxvcGVyQG15c3FsLmNvbTBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQC2j+W3tIaD
-E4r5v2PLZC25UdHeq3tFH6q1ZnMT+aYH1bp8+pK9N+Kth9s+tmoSZPjuF+MVBi+o
-gmi/V43DBJgnAgMBAAGjgaMwgaAwCQYDVR0TBAIwADAdBgNVHQ4EFgQU2Zq4XyLq
-BBDIJX2CV+Yu/Rkp59owdAYDVR0jBG0wa4AU8uLqVWWkmuKsnZf1RWz294wRrd+h
-SKRGMEQxCzAJBgNVBAYTAlNFMRAwDgYDVQQIEwdVcHBzYWxhMRAwDgYDVQQHEwdV
-cHBzYWxhMREwDwYDVQQKEwhNeVNRTCBBQoIJAJXpePU0UOTVMA0GCSqGSIb3DQEB
-BQUAA4GBAFQHLSELpa87WCMyXlZ/q1hjSJGqOJCJFvnMv6QOeCufxRtYpuYIjy6u
-lwMhm/HNwCaPHdcoJ6COgQkbHA/JpUE6LUQ/nPqH/8hMK0T3G8E+TwF/6SbMnxwG
-tQsn0RCQvpMMnOew0eonmU4GFAx66cFSxTNovGEN24E7V0hXv0Ka
+MIIDMTCCAhkCAQEwDQYJKoZIhvcNAQEFBQAwRDELMAkGA1UEBhMCU0UxEDAOBgNV
+BAgTB1VwcHNhbGExEDAOBgNVBAcTB1VwcHNhbGExETAPBgNVBAoTCE15U1FMIEFC
+MB4XDTEwMDIyMDA1MjQwNVoXDTEyMTExNjA1MjQwNVoweTELMAkGA1UEBhMCU0Ux
+EDAOBgNVBAgTB1VwcHNhbGExETAPBgNVBAoTCE15U1FMIEFCMRIwEAYDVQQDEwls
+b2NhbGhvc3QxMTAvBgkqhkiG9w0BCQEWImFic3RyYWN0Lm15c3FsLmRldmVsb3Bl
+ckBteXNxbC5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC/Ek3U
+eGMjK+fSNsYWjYe5Y+m+FmDLoK2J1dzVuWQ/cvE+piAvKmdGZNors0EqZUokpskx
+/RLr8VLqRHrV3iFJ6gElf6q1FBCX9lD1gyXWhdx5BjcLhmgQga0ZAQ1o/eE71PP6
++oNnojqMXcyaNmU2pxtVmcUSEdnx8nhyRgRlkH25sPcpl+nX/MdyTLIQrYcuHRyw
+n00HbvF606I/EbbTUWkXe2L7fIjHwUWDP59OvWkopvj7GJ2ppxvO/OMkqV/0gxoT
+30ieRQZXIRFH8ew03WK0sOJq0ZT993yRUA7yOpxPcCgoztIzuPqWX05Ow+dJiTJr
+cwVaura+qEJwNv3FAgMBAAEwDQYJKoZIhvcNAQEFBQADggEBABVprvNAuws1C4Lc
+Orjl154WR59OuxzJCUZrlw+8oKn4JrSf1vaRiF8IyrD4cwK4QOkOm/EjedcMjh49
+0j64gXU5a4oea1U0wVXF0SUqcIGOGP/+Tw+Oje+kv+Ydm5frfHoFFv+EE0lzuAqn
+KoQBYKWSf5Jx8Gb88xlaOqnRSRaPo1tdjtmEecD1D/z+Vvd1qZSvpnOPsiPiNEmy
+Rc0HNIyPO/99qee5SfIZ3eUKbs5nmB8tQgxEu6/wMQaIUjtNPpffYxKYNnLQUsVg
+aIcsYLz7yz6PfKoCogNAN45dZ1ZaHNfC7Lifq9IdqsBtfATGHLU2MOAI8jP1dOOF
+S1YMCEk=
 -----END CERTIFICATE-----
diff -Nur a/mysql-test/std_data/server-key.pem b/mysql-test/std_data/server-key.pem
--- a/mysql-test/std_data/server-key.pem	2010-02-20 00:28:37.000000000 -0500
+++ b/mysql-test/std_data/server-key.pem	2010-02-20 00:30:02.000000000 -0500
@@ -1,9 +1,27 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIIBOQIBAAJBALaP5be0hoMTivm/Y8tkLblR0d6re0UfqrVmcxP5pgfVunz6kr03
-4q2H2z62ahJk+O4X4xUGL6iCaL9XjcMEmCcCAwEAAQJASA5VwgNb0CKHiPm0ntOk
-hG+54SRX3DmafEy6gRjZIl/bZ/asSLhXUZ+CeohyrQh7BZgYWvykd8pRISL9eKsU
-GQIhAOXkUrOtP/EtjyqNluEqZdG+RZi/7p61JS3Ce13Myu+LAiEAy0uMlV34AJpM
-b40FPKqlHxw8DD/Dt1iKhNVAg8+LDVUCIFjv7fbJDbW2VG63/Cj8CAwOukoP5rbL
-iaicVrHBKrllAiB9+MiaXeopZXNrxDS0jQFYr8Q9yt1aJVFgUkxx4Q9HKQIgZPs0
-KlF3NNNWw78INaAEkyf0IEssnLMsuoCWw0DIOak=
+MIIEpQIBAAKCAQEAvxJN1HhjIyvn0jbGFo2HuWPpvhZgy6CtidXc1blkP3LxPqYg
+LypnRmTaK7NBKmVKJKbJMf0S6/FS6kR61d4hSeoBJX+qtRQQl/ZQ9YMl1oXceQY3
+C4ZoEIGtGQENaP3hO9Tz+vqDZ6I6jF3MmjZlNqcbVZnFEhHZ8fJ4ckYEZZB9ubD3
+KZfp1/zHckyyEK2HLh0csJ9NB27xetOiPxG201FpF3ti+3yIx8FFgz+fTr1pKKb4
++xidqacbzvzjJKlf9IMaE99InkUGVyERR/HsNN1itLDiatGU/fd8kVAO8jqcT3Ao
+KM7SM7j6ll9OTsPnSYkya3MFWrq2vqhCcDb9xQIDAQABAoIBAAnZ9L9Wsa3Xjq1M
++6ugAnZxi1SsrLLzBt+LckZmt5952qxjPleQNSDxBDqqlFqOPGNlYgw8wyP9HkxL
+2ZV7TXYVJZf8XB3X+f8LySLPbzz/S93Fj0L93BG1zenzHvDh3iIKmcW/8tR+Dcy8
+7VaqJU/LD+K6+ekzmGwcF0+1AncGBCpZhLQvuEzB3PNe4HJ38EUTNerIMjWktXRs
+jCG5cMbYGhvgoO5TjdKzfOjXBrAKUTXGZAFrbPWgLBYIcpHSE3u/p2PzBEuOIbEu
+10Ufe0QzkYC/O+/Wcg0yDJyJbdHvIMupgc8iSWgiy+F8dDX4A8J4/mWnFVlLC4iv
+Yg833gUCgYEA+kWhmE3v6AkJelbiDl5eqtInUzvCeA0OV6+gWMaOwvebmct/57B4
+dGrGIHPLjTIdv2Fj+6rJlAQkYMgTWSbb1wj2MrZwIQqTzbz9KzbpGPDmubEE/QG+
+o/wdRMXNQRjog7UacXUgoUdzzGGTg7wutQ/fRZJ9ctZtvSRxFXIyIZMCgYEAw3HP
+po0iejviRuB8m93q81BoamArxtKbw3PsGrR4wzKK7paNVo1FYAe5o9x4bMtBPqTR
++Bm6NleeBtlqIyl3+fhraAn9RAUmyPsTRcG6/PVpKLmr/u/TYq1yIrBPLt9itV7W
+s2VCap8hCmzVqc+0qHPxh7tgK2BIq2+4JGK2WkcCgYEA2aTuR8YXSQF65ZJjGj3F
+M+Z4XnDMyJwKCgVoe2uJQVIVViTbwX/88LjePeCVdt/LcOBgE79Nd9Q2UpgMA27A
+pftcJ3PyhTlume+tdwRUp+TvsTKZdPEsee1JAsLFocPoiIlZSquiSb8am/WLVHe1
+Yq4/63/eapzBaEgls5HQuNMCgYEAvSe1VPhbWfoDMZ2qfB/3eK1loS6ar769yhiz
+ddUc265bEDbda9smPZ68aH7EfmbpKnvSS1B7DIUYCID7XDSSTXfEW9jvKZniw7lc
+GbM6CgBqWmP4tq/d4i8Du9bvi+nfWrIDEXT7Jr4D+FcFI2hG7JHqPPd4SALT0jg5
++90dp5kCgYEAtwTjkhryQhMKwfy/WIThOV24eneMqSQ4a7mYq102GibOfdbYgZL9
+xJuaChrVZtLLC5w3wCiWO53F147Zl7OrTL099TOMgK2YtVZcHYYnXCkIEBSxHY+n
+mAjOSbY14UNQbR3Qz6OFAf3aNK5zM5pHbjce7Se9jXZ8k/haXW7jm3s=
 -----END RSA PRIVATE KEY-----
